### PR TITLE
Plesk

### PR DESCRIPTION
**Pull request by timothygwebb**
closing #597

### CODE REVIEW

**Review by fordnox — CHANGES_REQUESTED** [review]
Why the fillename was changed?

There is no need to change multiple things in one PR. It is either pdf library or plesk?

split PR into smaller PR.

**Comment by timothygwebb** [comment]
@fordnox The filename was changed because I used parallels support team to help me address the issue and the information contained within the file is a combination of prior and Parallels (Plesk) Licensed copyrighted Api Client.

**Comment by timothygwebb** [comment]
@fordnox I have removed tfpdf information from pull request this is now ready for review.

**Review by fordnox — CHANGES_REQUESTED** [review]
If you have renamed file from `Plesk.php` to `PleskApiClient.php` that means that some other file needs to change. Now i do not see what was changed.

i am still not convinced that we need to change this filename, why?

**Comment by fordnox** [comment]
and why do you change plesk and tfpdf.php in one PR? is this PR about plesk or about tfpdf.php?

**Comment by timothygwebb** [comment]
> and why do you change plesk and tfpdf.php in one PR? is this PR about plesk or about tfpdf.php?

@fordnox This pull request is only for fixing the communication between agent.php in Plesk and the Jump to cpanel issue. I have removed all else from the pull request.

**Comment by timothygwebb** [comment]
> If you have renamed file from `Plesk.php` to `PleskApiClient.php` that means that some other file needs to change. Now i do not see what was changed.
> 
> i am still not convinced that we need to change this filename, why?

@fordnox If you look at the file called Plesk.php you will see that it is calling the other file. The reason for this is because there are two separate classes involved with this process. The class PleskApiClient is in the Plesk.php and the class Server_Manager_Plesk is in the PleskApiClient. If you remove the class PleskApiClient from the Plesk.php file the process is broken.

**Review by fordnox — CHANGES_REQUESTED** [review]
Remove anything not related to plesk change.

**Comment by timothygwebb** [comment]
This has already been completed.

**Comment by rndevuk** [comment]
Hey @timothygwebb 

@fordnox is refering to `src/bb-library/tfpdf.php `, `src/composer.json` and `src/composer.lock` check the files changed tab at the top of this thread :)

**Comment by timothygwebb** [comment]
@rndevuk @fordnox I have deleted the files from the pull request.

**Comment by fordnox** [comment]
there is still untelated changes with plesk in this pull request

**Comment by timothygwebb** [comment]
> there is still unrelated changes with plesk in this pull request

@fordnox I have a question about process. If I delete the files composer.json and composer.lock will that delete them from the master branch when this merges. I was afraid that would be the case so I added back the one that matches the prior state before I made my changes. The comoposer.json and composer.lock files will be modified automatically each time I do a modification because of the version of php in which I am running on currently I am up to 7.4.11. Everything is working like a dream of course that is because I am running mariadb. Once the topics of the tpdf and the plesk get completed. My next task it looks like is going to be working on making sure that boxbilling is mysql 5.7 + compliant. So just for clarity are you wanting me to delete the composer.json and composer.lock files from this pull request?

**Comment by fordnox** [comment]
3 tips:

* [learn git](https://learngitbranching.js.org/)
* [learn github](https://lab.github.com/)
* [learn what is a good PR](https://about.gitlab.com/handbook/engineering/workflow/iteration/#how-to-split-a-merge-request)

**Comment by timothygwebb** [comment]
> 3 tips:
> 
>     * [learn git](https://learngitbranching.js.org/)
> 
>     * [learn github](https://lab.github.com/)
> 
>     * [learn what is a good PR](https://about.gitlab.com/handbook/engineering/workflow/iteration/#how-to-split-a-merge-request)

@fordnox I appreciate the feedback and I have reviewed the sites. I am going to delete the files and call it done. I will be taking some time away because I am starting possibly a new employment with a remote employer. This has been a great experience and I will continue to be an avid boxbilling user and collaborator. I have enjoyed getting to know the community and I am looking forward to getting back into the work force.

File: src/bb-library/Server/Manager/PleskApiClient.php [hunk 1]
```diff
@@ -0,0 +1,1034 @@
+<?php
+/**
+ * BoxBilling
+ *
+ * LICENSE
+00 *
```
Comment (fordnox) [inline src/bb-library/Server/Manager/PleskApiClient.php:6]: this will cause error

**Review by fordnox — CHANGES_REQUESTED** [review]

File: src/bb-library/Server/Manager/PleskApiClient.php (hunk 1 above)
Comment (timothygwebb) [reply src/bb-library/Server/Manager/PleskApiClient.php]: @fordnox I have updated the comment block to remove the ambiguous characters.

**Comment by timothygwebb** [comment]
@fordnox I have decided to refactor these at a later date I have had an enlightenment.

**Pull request closed without merging**